### PR TITLE
[JavaScript] Add Concept Exercise: type-conversion

### PR DESCRIPTION
**Pull request by shubhsk88**
The PR addresses issue #1892  which is the implementation of the new exercise for type-conversion in javascript

### CODE REVIEW

**Comment by SleeplessByte** [comment]
Started reviewing this. It's coming!

File: languages/javascript/exercises/concept/type-conversion/.docs/hints.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## 1. Calculate the sum of those two array
+
+- There is a built-in function to join the `array` and convert it to a string doing the implicit
+  type conversion to make it string
+- There is built in object which is explicitly used to convert the `string` to the `number`
+- Sum the result of the two converted number
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/type-conversion/.docs/hints.md:6]:
```suggestion
- Sum the result once each number is converted
```

File: languages/javascript/exercises/concept/type-conversion/.docs/hints.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+## 1. Calculate the sum of those two array
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/type-conversion/.docs/hints.md:1]:
```suggestion
## 1. Find the sum of two arrays
```

File: languages/javascript/exercises/concept/type-conversion/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+In this exercise you'll be writing code to understand the concepts of type conversion that you might encounter in your everydahy code
+
+You are given two array with number as element in it.You need to add those the sum the result of combining the number in the array
+
+## Tasks
+
+## Calculate the sum of those two array
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/type-conversion/.docs/instructions.md:7]:
```suggestion
## 1. Find the sum of two arrays
```

File: languages/javascript/exercises/concept/type-conversion/.docs/hints.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+- There is a built-in function to join the `array` and convert it to a string doing the implicit
+  type conversion to make it string
+- There is built in object which is explicitly used to convert the `string` to the `number`
+- Sum the result of the two converted number
+
+## 2. Calculate the function which tells us the owner whether the number is lucky or not
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/type-conversion/.docs/hints.md:8]:
```suggestion
## 2. Determine if the number is lucky
```

File: languages/javascript/exercises/concept/type-conversion/.docs/instructions.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+//=> 130
+```
+
+The result should be a number and the array element only contains as integer
+
+After the calculator the owner would like to know the sum that he got is a lucky number or not.
+
+A number is said to be lucky if the the reverse of the number is the same
+
+## 2. Calculate the function which tells us the owner whether the number is lucky or not
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/type-conversion/.docs/instructions.md:22]:
```suggestion
## 2. Determine if the number is lucky
```

File: languages/javascript/exercises/concept/type-conversion/.docs/hints.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+- There is a prototype which explicit convert the number to `string`
+- There is a function which can convert the `string` to the `array`.
+- There is a function on the `array` constructor which can reverse the number of elements in your array
+- First convert the number into the string and then convert it into array and calculate the reverse of it and join the array and finally compare the two string
+
+## 3. Dashify the number
+
+- Convert the number into string and then convert it into the array and finally join the string using the function to `join` the array using dash
```
Comment (neenjaw) [inline languages/javascript/exercises/concept/type-conversion/.docs/hints.md:17]: Try not to give away the solution with the hint.

```suggestion
- Use type conversion to change the number to a string
- Strings have methods which can be used to break them apart
- Arrays have a method to merge elements into a string
```

You could apply this to each hint as well

**Review by neenjaw — CHANGES_REQUESTED** [review]
I think the only change that _must_ be done is changing the after.md into a concept about.md with a links.json.

The rest is pretty good (:tada: :tada:) and if you don't want to make the changes, it should have an improvement issue opened.

@SleeplessByte is busy, but it would also  be good to have his eye on this as well.

Great work!

File: languages/javascript/exercises/concept/type-conversion/.docs/after.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+The main thing to remember while doing Javascript type-conversion that we can convert type in multiple way for different objects and explicit type-conversion operation such as `Array` and using unary operator `+` .
+
+You can find the all the methods in [MDN docs][mdn docs] or [ECMA docs][ecma docs]
+
+[mdn docs]: https://developer.mozilla.org/en-US/docs/Mozilla/js-ctypes/Using_js-ctypes/Type_conversion
+[ecma docs]: https://www.ecma-international.org/ecma-262/#sec-abstract-operations
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply languages/javascript/exercises/concept/type-conversion/.docs/after.md:6]: I agree with this.

File: languages/javascript/exercises/concept/type-conversion/.docs/instructions.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+Implement a function to determine whether the number is lucky for him or not.
+
+```javascript
+luckyNumber(1441)
+//=>  true
+luckyNumber(123)
+//=> false
+```
+
+He decide to send the result of this number to his partner to make it easy for him to read
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/instructions.md:33]:
```suggestion
```

File: languages/javascript/exercises/concept/type-conversion/.docs/instructions.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+## 3. Dashify the number
+
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/instructions.md:36]:
```suggestion

He decides to send the result of this number to his partner to make it easy for him to read
```

File: languages/javascript/exercises/concept/type-conversion/.docs/instructions.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+twoSum([1, 2, 3], [0, 0, 7])
+//=> 130
+```
+
+The result should be a number and the array element only contains as integer
+
+After the calculator the owner would like to know the sum that he got is a lucky number or not.
+
+A number is said to be lucky if the the reverse of the number is the same
+
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/instructions.md:21]:
```suggestion

```

File: languages/javascript/exercises/concept/type-conversion/.docs/instructions.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+## 2. Calculate the function which tells us the owner whether the number is lucky or not
+
+Implement a function to determine whether the number is lucky for him or not.
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/instructions.md:24]:
```suggestion
After the calculator the owner would like to know the sum that he got is a lucky number or not.

A number is said to be lucky if the the reverse of the number is the same

Implement a function to determine whether the number is lucky for him or not.
```

File: languages/javascript/exercises/concept/type-conversion/.docs/introduction.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
+Type Conversion is often referred when converting the value from one data type to another data type.This is also often referred as "Coercion".
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/introduction.md:1]:
```suggestion
Type Conversion is the act of converting the value from one data type to another data type. This is also often referred as "Coercion".
```

File: languages/javascript/exercises/concept/type-conversion/.docs/introduction.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+There are two ways to do the type conversion
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/introduction.md:3]:
```suggestion
Type conversion may be explicit or implicit.
```

File: languages/javascript/exercises/concept/type-conversion/.docs/introduction.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+The first is the explicit conversion which is usually those type conversions that are obvious and
+explicit:
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/introduction.md:6]:
```suggestion
Explicit type conversion are performed by a function call _meant to convert the type_:
```

File: languages/javascript/exercises/concept/type-conversion/.docs/introduction.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+String(a)
+// => "46"
+const b = '3.4'
+Number(b)
+//=> 3.4
+Boolean(b)
+// => true
+```
+
+The other one is the implicit conversion are type conversion which is usually hidden and they are more non-obvious
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/introduction.md:19]:
```suggestion
Implicit conversion happens as a side-effect of an operation:
```

File: languages/javascript/exercises/concept/type-conversion/.docs/introduction.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+```javascript
+const a = 10
+console.log(`There are ${a} cats in the house`) // Number=>String
```
Comment (SleeplessByte) [inline languages/javascript/exercises/concept/type-conversion/.docs/introduction.md:23]:
```suggestion
console.log(`There are ${a} cats in the house`) 
// Implicitly converts from Number to String in order to insert the value of a into the template literal.
```

File: languages/javascript/exercises/concept/type-conversion/.docs/introduction.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+a.getTime() // Date object to number
+
+const b = { pokemon: 'pikachu' }
+
+const c = JSON.stringify(b) //Object to string
+//=> '{"pokemon":"pikachu"}'
+
+JSON.parse(c) //string to object
+//=>{pokemon:"pikachu"}
+```
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply languages/javascript/exercises/concept/type-conversion/.docs/introduction.md:47]: They are type-conversions. They're not what's traditionally known as "casts".

**Comment by SleeplessByte** [comment]
@shubhsk88 hows' this going?

**Comment by shubhsk88** [comment]
> @shubhsk88 hows' this going?

Sorry for the delay .Bro this and the other  abstract equality one will be done by this weekend .got busy with some work but will do it

**Review by SleeplessByte — DISMISSED** [review]
I think something went wrong here. It's changing almost every file from every track. Did you attempt to rebase?

**Comment by shubhsk88** [comment]
> I think something went wrong here. It's changing almost every file from every track. Did you attempt to rebase?

I think there was some change in the the way exercise were organised and when I tried to pull it,it just have too much changes could you help me to do that @SleeplessByte

**Comment by SleeplessByte** [comment]
It's probably easiest to do this:

1. Grab the new exercise directory
2. Put it somewhere safe
3. Delete the branch
4. Make sure main/master is up to date
5. Create branch with the same name you just deleted
6. Put back the exercise directory
7. Commit it
8. Force push (same upstream)

**Comment by neenjaw** [comment]
PR Fixed :P

**Review by neenjaw — APPROVED** [review]

**Pull request merged**